test/basics/test_linalg.jl

```diff
@@ -0,0 +1,35 @@
+using SparseArraysBase: SparseArrayDOK
+using LinearAlgebra: mul!
+using Random
+
+# TODO: add this to main package
```

Comment:
Good idea, though as I'm sure you know we should make it more general, i.e. allow specifying the axes or size and allow passing an RNG.

Reply:
Haha indeed, this is why I didn't add it immediately. 😄  I'll get to it soon, just want to keep the PRs clean so they have a single topic, making it a lot easier to review and the git history is better